Node tries to process request before it is ready to do so

neofs-node v0.25.1 (Should relevant for 0.26.0 too)

Expected behavior: do not process any requests during startup routine.
Actual behavior: node tries to process `LocalNodeInfo` request before node is ready and throws panic because network state field is not set yet.

```
2021-10-20T08:59:10.692Z        info        event/listener.go:485        registered new event handler        {"contract": "9dbd2b5e67568ed285c3d6f96bac4edf5e1efba0", "event_type": "StopEstimation"}
2021-10-20T08:59:10.692Z        info        event/listener.go:448        registered new event parser        {"contract": "9dbd2b5e67568ed285c3d6f96bac4edf5e1efba0", "event_type": "StartEstimation"}
2021-10-20T08:59:10.692Z        info        event/listener.go:485        registered new event handler        {"contract": "9dbd2b5e67568ed285c3d6f96bac4edf5e1efba0", "event_type": "StartEstimation"}
2021-10-20T08:59:10.693Z        info        neofs-node/grpc.go:60        start listening gRPC endpoint        {"endpoint": "65.21.109.202:8080"}
2021-10-20T08:59:10.698Z        info        neofs-node/grpc.go:60        start listening gRPC endpoint        {"endpoint": "65.21.109.202:8082"}
panic: interface conversion: interface {} is nil, not *netmap.NodeInfo
goroutine 212 [running]:
main.(*networkState).getNodeInfo(...)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/netmap.go:66
main.(*cfg).LocalNodeInfo(0xc000106300, 0x1070e50, 0xe4f640, 0x183c280)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/config.go:450 +0xc6
github.com/nspcc-dev/neofs-node/pkg/services/netmap.(*executorSvc).LocalNodeInfo(0xc0017ff800, 0x1184938, 0xc0002fe450, 0xc001898e58, 0x1177960, 0x183a4b0, 0xc00001f928)
        github.com/nspcc-dev/neofs-node/pkg/services/netmap/executor.go:51 +0x55
github.com/nspcc-dev/neofs-node/pkg/services/netmap.(*responseService).LocalNodeInfo.func1(0x1184938, 0xc0002fe450, 0xfa13e0, 0xc001898e58, 0x0, 0x0, 0x40e2f8, 0x8)
        github.com/nspcc-dev/neofs-node/pkg/services/netmap/response.go:29 +0x69
github.com/nspcc-dev/neofs-node/pkg/services/util/response.(*Service).HandleUnaryRequest(0xc00000e0b8, 0x1184938, 0xc0002fe450, 0xfa13e0, 0xc001898e58, 0xc00001f9a8, 0x0, 0x118f0a8, 0x1180d38, 0x7f089fa9b580)
        github.com/nspcc-dev/neofs-node/pkg/services/util/response/unary.go:13 +0x5e
github.com/nspcc-dev/neofs-node/pkg/services/netmap.(*responseService).LocalNodeInfo(0xc0017e7818, 0x1184938, 0xc0002fe450, 0xc001898e58, 0xc001898e58, 0xc0002ff410, 0x0)
        github.com/nspcc-dev/neofs-node/pkg/services/netmap/response.go:27 +0x85
github.com/nspcc-dev/neofs-node/pkg/services/netmap.(*signService).LocalNodeInfo.func1(0x1184938, 0xc0002fe450, 0xfa13e0, 0xc001898e58, 0x40db01, 0xc0018bc420, 0xc00001fa88, 0xafb677)
        github.com/nspcc-dev/neofs-node/pkg/services/netmap/sign.go:29 +0x69
github.com/nspcc-dev/neofs-node/pkg/services/util.(*SignService).HandleUnaryRequest(0xc00000f070, 0x1184938, 0xc0002fe450, 0xfa13e0, 0xc001898e58, 0xc00001fad0, 0x0, 0x0, 0x18, 0xf1fd00) 
        github.com/nspcc-dev/neofs-node/pkg/services/util/sign.go:119 +0x114
github.com/nspcc-dev/neofs-node/pkg/services/netmap.(*signService).LocalNodeInfo(0xc0017e7830, 0x1184938, 0xc0002fe450, 0xc001898e58, 0x0, 0xece440, 0xc0002fe400)
        github.com/nspcc-dev/neofs-node/pkg/services/netmap/sign.go:27 +0x85
github.com/nspcc-dev/neofs-node/pkg/network/transport/netmap/grpc.(*Server).LocalNodeInfo(0xc001812540, 0x1184938, 0xc0002fe450, 0xc0003d6d40, 0xc001812540, 0xc0002fe450, 0xc00051dba0)
        github.com/nspcc-dev/neofs-node/pkg/network/transport/netmap/grpc/service.go:33 +0xa4
github.com/nspcc-dev/neofs-api-go/v2/netmap/grpc._NetmapService_LocalNodeInfo_Handler(0xece440, 0xc001812540, 0x1184938, 0xc0002fe450, 0xc0018a0960, 0x0, 0x1184938, 0xc0002fe450, 0xc0000b2dc0, 0x147)
        github.com/nspcc-dev/neofs-api-go@v1.29.0/v2/netmap/grpc/service.pb.go:869 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001efdc0, 0x118e3d8, 0xc000365b00, 0xc001810240, 0xc0017ff890, 0x17f1f40, 0x0, 0x0, 0x0) 
        google.golang.org/grpc@v1.38.0/server.go:1286 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0001efdc0, 0x118e3d8, 0xc000365b00, 0xc001810240, 0x0)
        google.golang.org/grpc@v1.38.0/server.go:1609 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0019ad7c0, 0xc0001efdc0, 0x118e3d8, 0xc000365b00, 0xc001810240)
        google.golang.org/grpc@v1.38.0/server.go:934 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,9 @@ Changelog for NeoFS Node
 
 ## [Unreleased]
 
+### Fixed
+- Storage Node handles requests before its initialization is finished (#934)
+
 ## [0.26.0] - 2021-10-19 - Udo (우도, 牛島)
 
 NeoFS API v2.10 support

--- a/cmd/neofs-node/netmap.go
+++ b/cmd/neofs-node/netmap.go
@@ -90,6 +90,8 @@ func initNetmapService(c *cfg) {
 		initMorphComponents(c)
 	}
 
+	initNetmapState(c)
+
 	server := netmapTransportGRPC.New(
 		netmapService.NewSignService(
 			&c.key.PrivateKey,
@@ -162,9 +164,9 @@ func initNetmapService(c *cfg) {
 	}
 }
 
+// bootstrapNode adds current node to the Network map.
+// Must be called after initNetmapService.
 func bootstrapNode(c *cfg) {
-	initState(c)
-
 	if c.needBootstrap() {
 		err := c.bootstrap()
 		fatalOnErrDetails("bootstrap error", err)
@@ -191,7 +193,9 @@ func setNetmapNotificationParser(c *cfg, sTyp string, p event.NotificationParser
 	c.cfgNetmap.parsers[typ] = p
 }
 
-func initState(c *cfg) {
+// initNetmapState inits current Network map state.
+// Must be called after Morph components initialization.
+func initNetmapState(c *cfg) {
 	epoch, err := c.cfgNetmap.wrapper.Epoch()
 	fatalOnErrDetails("could not initialize current epoch number", err)
 

